tests/FSharpLint.Core.Tests/Rules/Conventions/Naming/NamingHelpers.fs

```diff
@@ -17,3 +19,12 @@ type TestConventionsNamingHelpers() =
         Assert.IsTrue(isCamelCase "dogInBin")
 
         Assert.IsFalse(isCamelCase "DogInBin")
+
+    [<Test>]
+    member __.GetPatternIdentsRegression() =
+        let lhsPat = SynPat.Null Range.Zero
+        let rhsPat = SynPat.Named(SynIdent(Ident("foo", Range.Zero), None), false, None, Range.Zero)
+        let pattern = SynPat.As(lhsPat, rhsPat, Range.Zero)
+        let result = getPatternIdents AccessControlLevel.Public (fun _ synPat -> Array.singleton synPat) false pattern
+
+        CollectionAssert.AreEqual(result, Array.singleton rhsPat)
```

Comment:
@webwarrior-ws why would you add a test that is COMPLETELY DIFFERENT from the other ones in the test suite? I of course meant a test with source code snippet

Reply:
Other tests that use source code are based on a specific rule. But here a helper function is tested.

Reply:
but this bug happened in a specific rule as well; there's no reason to not have a snippet-based test here

Reply:
Made a test in ParameterNames tests that uses source code snippet instead of the old one.

Reply:
Thanks! And just to reiterate about my decision above: all of the tests that are in the Rules/ folder are meant to be end-to-end tests, not unit tests of particular APIs.